### PR TITLE
docs: update some rspack legacy descriptions

### PR DESCRIPTION
**Pull request by 9aoy**
## Require Documentation?

- [ ] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

File: website/docs/zh/guide/glossary.mdx [hunk 1]
```diff
@@ -28,7 +28,7 @@ Code splitting 是一种技术，它允许你将你的代码拆分成多个块
 
 ## first class module type（一等公民模块类型）
 
-Rspack 中的一等公民模块指的是那些不需要依赖 loader 和 plugin 即可支持的模块类型，例如 JavaScript、CSS、JSON 等，而像 HTML、Markdown、YAML 等需要依赖 loader 和 plugin 才能支持的模块类型则不是一等公民模块。
+Rspack 中的一等公民模块指的是那些不需要依赖 loader 和 plugin 即可支持的模块类型，例如 CSS、JSON、资源等，而像 JavaScript、HTML、Markdown、YAML 等需要依赖 loader 和 plugin 才能支持的模块类型则不是一等公民模块。
```
Comment (h-a-n-a) [inline website/docs/zh/guide/glossary.mdx:31]: 符合 Web 标准的 JavaScript 其实也算是一等公民

File: website/docs/zh/guide/glossary.mdx [hunk 2]
```diff
@@ -28,7 +28,7 @@ Code splitting 是一种技术，它允许你将你的代码拆分成多个块
 
 ## first class module type（一等公民模块类型）
 
-Rspack 中的一等公民模块指的是那些不需要依赖 loader 和 plugin 即可支持的模块类型，例如 JavaScript、CSS、JSON 等，而像 HTML、Markdown、YAML 等需要依赖 loader 和 plugin 才能支持的模块类型则不是一等公民模块。
+Rspack 中的一等公民模块指的是那些不需要依赖 loader 和 plugin 即可支持的模块类型，例如 CSS、JSON、资源等，而像 TypeScript、HTML、Markdown、YAML 等需要依赖 loader 和 plugin 才能支持的模块类型则不是一等公民模块。
```
Comment (h-a-n-a) [inline website/docs/zh/guide/glossary.mdx:31]:
```suggestion
Rspack 中的一等公民模块指的是那些不需要依赖 loader 和 plugin 即可支持的模块类型，例如 JavaScript、CSS、JSON、资源等，而像 TypeScript、HTML、Markdown、YAML 等需要依赖 loader 和 plugin 才能支持的模块类型则不是一等公民模块。
```

File: website/docs/en/guide/glossary.mdx [hunk 3]
```diff
@@ -36,7 +36,7 @@ Code splitting is a technique that allows you to split your code into multiple c
 
 ## first class module type
 
-In Rspack, first-class module types refer to module types that can be supported without relying on loaders or plugins, such as JavaScript, CSS, JSON, etc. However, module types that require loaders or plugins to support, such as HTML, Markdown, YAML, etc., are not first-class citizen modules.
+In Rspack, first-class module types refer to module types that can be supported without relying on loaders or plugins, such as CSS, JSON, Assets, etc. However, module types that require loaders or plugins to support, such as TypeScript, HTML, Markdown, YAML, etc., are not first-class citizen modules.
```
Comment (h-a-n-a) [inline website/docs/en/guide/glossary.mdx:39]:
```suggestion
In Rspack, first-class module types refer to module types that can be supported without relying on loaders or plugins, such as JavaScript, CSS, JSON, Assets, etc. However, module types that require loaders or plugins to support, such as TypeScript, HTML, Markdown, YAML, etc., are not first-class citizen modules.
```

File: website/docs/zh/guide/glossary.mdx (hunk 1 above)
Comment (9aoy) [reply website/docs/zh/guide/glossary.mdx]: 👍

**Review by h-a-n-a — APPROVED** [review]

**Pull request merged**